src/main/java/de/tum/in/www1/artemis/service/connectors/localci/SharedQueueManagementService.java

```diff
@@ -296,13 +301,22 @@ public Page<BuildJob> getFilteredFinishedBuildJobs(FinishedBuildJobPageableSearc
         Duration buildDurationLower = search.buildDurationLower() == null ? null : Duration.ofSeconds(search.buildDurationLower());
         Duration buildDurationUpper = search.buildDurationUpper() == null ? null : Duration.ofSeconds(search.buildDurationUpper());
 
+        var sortOptions = Sort.by(search.pageable().getSortedColumn());
+        sortOptions = search.pageable().getSortingOrder() == SortingOrder.ASCENDING ? sortOptions.ascending() : sortOptions.descending();
```

Comment:
Together with the next suggestion it simplifies the sorting:
```suggestion
var sortOptions = Sort.by(search.pageable().getSortedColumn());
        var comparator = Comparator.comparingLong(BuildJob::getId);
        if (search.pageable().getSortingOrder() == SortingOrder.ASCENDING) {
            sortOptions = sortOptions.ascending();
        } else {
            sortOptions = sortOptions.descending();
            comparator = comparator.reversed();
        }
```

Reply:
This won't really work as we dont want the buildJobs to be sorted based on Ids, but want them to keep the same order as the result of the first fetch